### PR TITLE
JACE benchmarking: Dockerfile update

### PR DESCRIPTION
**Pull request by sbesson**
Follow-up of https://github.com/openmicroscopy/ome-files-performance/pull/5, this PR updates the Dockerfile in order to be able to build and execute the JACE bindings in the Ubuntu 16.04 environment.

A couple of changes are still in progress:
- the base image should be replaced by the organization one once https://github.com/sbesson/ome-files-cpp-u1604-docker is updated to build the current OME Files
- the benchmarking scripts will execute all 3 components (OME Files C++, JACE, Bio-Formats)
- modifies `javaTools.cpp` to add the JAR folder to the CLASSPATH and detect the logging configuration file (could be replaced by using another signature of `createJVM` with extra classpath)

### CODE REVIEW

**Comment by sbesson** [comment]
@rleigh-codelibre with the last set of commits, I can build the various components but I encounter failures at runtime of type

```
root@af3155d9ad6b:/git/ome-files-performance# ./scripts/run_benchmarking 
rm: cannot remove '/data/results/*': No such file or directory
rm: cannot remove '/data/out/*': No such file or directory
Exception creating JVM: Assert failed: Unable to find the class, jace.util.ShutdownHook.
WARNING in native method: JNI call made with exception pending
terminate called after throwing an instance of 'jace::proxy::java::lang::Throwable'
  what():  java.lang.NoClassDefFoundError: loci/formats/services/OMEXMLServiceImpl
Aborted (core dumped)
```

Probable something related to the environment. Will investigate tomorrow unless you have some ideas.

**Comment by rleigh-codelibre** [comment]
@sbesson You do need to copy the `jar` directory from the jace build into the current working directory when running the tests.  (There may be a cleaner way of doing things, but this is what I found to work.)

**Comment by sbesson** [comment]
With the latest commits, I can successfully run the JACE tests in the `run_benchmarking` scripts against the two filesets (BBBC and mitocheck). Next todo: try to fix the logging initialization which is currently printed unconditionally at DEBUG level.

**Comment by sbesson** [comment]
Last commit includes an update of the statistics notebook with the latest results of the Linux/Windows builds using a recent version of the development OME Files. A few notes:

- this does not include the recent C++ performance improvements (https://github.com/ome/ome-files-cpp/pull/54 and https://github.com/ome/ome-model/pull/28)
- in most cases, the JACE times are similar to the Java times. The only notable exception is the BBBC writing where JACE is significantly slower. Is that a scaling limitation related to the ByteArray handling?
- for BBBC the raw JACE pixels table only includes 1 recorded point  and for mitocheck the JACE generated OME-TIFF is 3 times larger than the Java/C++ equivalent. I suspect we need to delete the file at the beginning of the test similarly to https://github.com/openmicroscopy/ome-files-performance/commit/5fd90c5ef29ba8641be7fa3f9b4c1c3f7eddee8a.

File: CMakeLists.txt [hunk 1]
```diff
@@ -64,7 +64,7 @@ find_package(OMECompat 5.2.0 REQUIRED)
 find_package(OMECommon 5.2.0 REQUIRED)
 find_package(OMEXML 5.2.0 REQUIRED)
 find_package(OMEFiles 0.2.3 REQUIRED)
-find_package(BioFormatsJACE 5.2.4)
```
Comment (rleigh-codelibre) [inline CMakeLists.txt:67]: Didn't we need to revert this to get the correct header versioning?

File: Dockerfile [hunk 2]
```diff
@@ -1,23 +1,49 @@
...
+ENV JAVA_HOME /usr/lib/jvm/java-7-openjdk-amd64
 
-ADD . /git/ome-files-performance
+#  Bio-Formats JACE bindings
+RUN git clone https://github.com/ome/bio-formats-jace /git/bio-formats-jace
+WORKDIR /git/bio-formats-jace
+RUN mvn -DskipTests clean package cppwrap:wrap dependency:copy-dependencies
 
-WORKDIR /build2
+COPY cmake/JACEPrerequisites.cmake /git/bio-formats-jace/target/cppwrap
```
Comment (rleigh-codelibre) [inline Dockerfile:17]: Needs renaming as well; the target is `/git/bio-formats-jace/target/cppwrap/Prerequisites.cmake`?

File: CMakeLists.txt (hunk 1 above)
Comment (sbesson) [reply CMakeLists.txt:67]: I think the issue was related to the JACE discovery and the `CMAKE_PATH_PREFIX` variable. After removing the `sed` workaround, I was able to rebuild everything and execute the JACE scripts.

**Comment by rleigh-codelibre** [comment]
Mostly looks fine I think; a couple of points noted.

**Comment by sbesson** [comment]
Discussed with @rleigh-codelibre. A couple of issues noted:
- https://github.com/openmicroscopy/ome-files-performance/pull/12#discussion_r100023917 was mostly required for Windows (not working), I will remove the copy step in a follow-up PR
- similar to Java, we need to remove the file before running the pixels performance tests if it exists. currently, the BBBC tests crash as the data is appended to the existing TIFF and overcomes the BigTIFF limit.
- additionally I will add a `set -e` step in the benchmarking script to fail on error

**Pull request merged**